### ISSUE TITLE
Static configuration in INFILE should be considered

### ISSUE THREAD

**Issue by ludamad** [hashkat/hashkat#31]
Hi, I talked to Kevin about this a while back. There is a way for the INFILE.yaml to control features in config_static.h, however this requires a compiler to be available. Given that we intend to distribute binaries, this feature would not be useful to anyone who only has a binary available, however. 

Basically the problem currently is that the compiler does well knowing at compile-time information such as the amount of preference classes, agent types, etc. Currently, we estimate an upper bound to these. 

There are two solutions:
- Full solution. Refactor code to not require static information. This is probably the most flexible going forward, but needs some rethinking. I like this option in the long-term. There should be no performance penalty if done correctly, but doing it correctly increases code complexity.
- Partial solution. If a compiler is available, use INFILE.yaml to aid compilation. This will mean that changing an INFILE can cause a recompilation of all of hashkat. As a long-running simulator, this is pretty negligible. It does, however, basically make a C++ compiler a runtime dependency (unless a .generated file is reused constantly)

**Comment by itamblyn** [comment]
I agree that the full solution is optimal, however, I’m not convinced it is worth the effort at this point.

What is the consequence of simply upping our estimates for some of these values (e.g. 100 languages for example) and add a note in the documentation that going beyond this number is straightforward, but not provided in the default binary.

Assuming there is no overhead cost to do this, I recommend static, “compile time options” for the moment, and we can revisit this in the future if it really becomes and issue.

**Comment by ludamad** [comment]
The overhead cost of putting eg 100 when you need 5 is currently very large
for certain things such as # of pref classes, due to follower list
partitioning. I'll dig deeper later to see if we can get rid of other ones.

**Comment by ludamad** [comment]
Languages are hard coded completely, by the way
